### PR TITLE
Addition GDWPS dataset wave data

### PR DESCRIPTION
**Pull request by prodyut1978**
## Background
Addition of new GDWPS wave dataset in the Ocean Navigator UI . The details description of the dataset is below: 
The Global Deterministic Prediction System (GDPS) performs physical calculations to produce deterministic forecasts of atmospheric elements from the present day to 10 days in the future. Elements of the atmosphere include temperature, precipitation, cloud cover, wind speed and direction, and humidity. This product contains the raw numerical results of these calculations. The geographical coverage is global. The horizontal spatial resolution of the data is 15 km. Data is available for 33 vertical levels. The forecasts are carried out twice a day.

## Why did you take this approach?
The client requested to add the Global Deterministic Prediction System (GDPS) wave dataset to our Ocean Navigator UI.

## Anything in particular that should be highlighted?
- Enable and Disable Quiver Variable selector for Mercator and NEMO/Fvcom Model. 
- Added and set "vector_arrow_stride": 10  in the `datasetconfig.json` file for high dense dataset (e.g. CMEMS)
- GDWPS Dataset is in [GRIB2](https://eccc-msc.github.io/open-data/msc-data/nwp_gdwps/readme_gdwps-datamart_en/) format. A script is built to convert it in the netCDF format. 

## Screenshot(s)

Capture_1

## Checks
- [Y ] I ran unit tests.
- [ ] I've tested the relevant changes from a user POV.
- [ ] I've formatted my Python code using `black .`.

_Hint_ To run all python unit tests run the following command from the root repository directory:
```sh
bash run_python_tests.sh
```

### CODE REVIEW

File: routes/api_v1_0.py [hunk 1]
```diff
@@ -469,10 +470,14 @@ def get_data_v1_0():
 
         bearings = None
         if "mag" in result["variable"]:
+            bearings_var = DatasetConfig(
+                result["dataset"]).variable[result["variable"]].bearing_component
```
Comment (htmlboss) [inline routes/api_v1_0.py:474]: You can use the `DatasetConfig` object that's already been instantiated on line 447!
```suggestion
            bearings_var = config.variable[result["variable"]].bearing_component
```

File: routes/api_v1_0.py [hunk 2]
```diff
@@ -469,10 +470,14 @@ def get_data_v1_0():
...
+            if not bearings_var:
+                bearings_var = "bearing"
```
Comment (htmlboss) [inline routes/api_v1_0.py:476]: You can use the `DatasetConfig` object that's already been instantiated on line 447! And you can also make use of how `or` short-circuits in Python to make the if-statement a one-liner:
```suggestion
            bearings_var = config.variable[result["variable"]].bearing_component or "bearing"   
```

Comment (htmlboss) [reply routes/api_v1_0.py:476]: So if `config.variable[result["variable"]].bearing_component` is `None`, "bearing" will be assigned to `bearings_var`.

File: oceannavigator/frontend/src/components/DatasetSelector.jsx [hunk 3]
```diff
@@ -370,7 +370,8 @@ class DatasetSelector extends React.Component {
     let quiverSelector = null;
     if (this.props.showQuiverSelector && !this.state.loading) {
       let quiverVariables = [];
-      if (this.state.datasetVariables) {
+      const ListvalidQuivModelClass = ["Mercator"]    // List of Valid Model_Class for Quiver Plot. Need to update this List when Model_Class : "Nemo" and "Fvcom" working
+      if ((ListvalidQuivModelClass.includes(this.state.model_class)) && (this.state.datasetVariables)) {
```
Comment (htmlboss) [inline oceannavigator/frontend/src/components/DatasetSelector.jsx:374]: Just a minor variable name tweak:

* Non-constant variables in JS should be defined using camelCase.

```suggestion
      const modelClassesWithQuiver = ["Mercator"]    // List of Valid Model_Class for Quiver Plot. Need to update this List when Model_Class : "Nemo" and "Fvcom" working
      if (this.state.datasetVariables && modelClassesWithQuiver.includes(this.state.model_class)) {
```

Also, `modelClassesWithQuiver` is a constant. I think this should be defined elsewhere in this file. Perhaps after `PARENT_ATTRIBUTES_TO_UPDATE`?

```js
const PARENT_ATTRIBUTES_TO_UPDATE = Object.freeze([
  "dataset",
  "dataset_attribution",
  "quantum",
  "variable",
  "variable_scale", // Default range values for variable
  "depth",
  "time",
  "starttime",
  "quiverVariable",
]);

const MODEL_CLASSES_WITH_QUIVER = Object.freeze(["Mercator"]);
```

File: oceannavigator/frontend/src/components/Defaults.js [hunk 4]
```diff
@@ -1,6 +1,7 @@
 
 const DATASET_DEFAULTS = Object.freeze({
   dataset: "giops_day",
+  model_class: "Mercator",
```
[the remark this answers is not in the document]
Comment (prodyut1978) [reply oceannavigator/frontend/src/components/Defaults.js:4]: Hi nabil, I changed the file accordingly. Thanks, Prodyut

**Pull request closed without merging**